### ISSUE TITLE
SSR not showing any warning for unknown tag

### ISSUE THREAD

**Issue by IlyaSemenov** [vuejs/vue#6998]
### Version
2.5.3

### Reproduction link
[https://jsfiddle.net/semenov/1oa635w8/](https://jsfiddle.net/semenov/1oa635w8/)

### Steps to reproduce
Run jsfiddle and note the warning message that is printed to the console.

Then run the same code with SSR:

```js
const Vue = require('vue')
const app = new Vue({
  template: '<div><bing-bang/></div>',
})

const renderer = require('vue-server-renderer').createRenderer()

renderer.renderToString(app, (err, html) => {
  if (err) throw err
  console.log(html)
})
```

Like this: `DEBUG=* node test.js`

### What is expected?
The SSR rendering prints a warning similar to what client-side rendering emits:

```
[Vue warn]: Unknown custom element: <bing-bang> - did you register the component correctly? For recursive components, make sure to provide the "name" option.

(found in <Root>)
```

### What is actually happening?
SSR simply renders invalid HTML:

```html
<div data-server-rendered="true"><bing-bang></bing-bang></div>
```

without any warnings.

---
This inconsistency makes it hard to refactor big Nuxt.js apps, as missing/unregistered components get silently ignored.

**Comment by LinusBorg** [comment]
Doesn't the warning appear on the client anyway?

**Comment by IlyaSemenov** [comment]
No, at least not on the initial SSR-generated page load (with Nuxt). It only shows a warning on the client when you arrive at the page with `<nuxt-link>`.

**Comment by yyx990803** [comment]
FYI It is expected behavior to not warn on the server, because it's possible that the user wants to render a real custom element instead of a Vue component. The warning can only be safely performed on the client where we can check if the element is registered as a real custom element.

Previously it is not warning even on client-side hydration, which is fixed in df82aeb.
